File: node/test/api/arch/dept.test.js
/**
 * Copyright (c) 2015-2022, FineX, All Rights Reserved.
 * @author Thomas.Zhuang
 * @date 2022/05/18
 */

import { DeptApiClient } from '../../../src/api/arch/dept';
import { API_KEY, HOST } from '../../../src/constants/http_constant';

export const deptTest = 'deptTest';

let deptNo;

describe('dept api test', () => {
    const api = new DeptApiClient(API_KEY, HOST, 'v5');

    test('deptCreate', async () => {
        const dept = await api.deptCreate('Demo研发部门', {});
        console.log('deptCreate result:' + JSON.stringify(dept));
        expect(dept['department']).toBeTruthy();
        deptNo = dept['department']['dept_no'];
    });


    test('deptList', async () => {
        const deptList = await api.deptList(deptNo, {
            hasChild: true
        });
        console.log('deptList result:' + JSON.stringify(deptList));
        expect(deptList.departments).toBeTruthy();
    });


    test('deptUpdate', async () => {
        const name = 'Demo测试部门';
        const dept = await api.deptUpdate(deptNo, name);
        console.log('deptUpdate result:' + JSON.stringify(dept));
        expect(dept['department']['dept_no']).toEqual(deptNo);
        expect(dept['department']['name']).toEqual(name);
    });

    test('deptDelete', async () => {
        const response = await api.deptDelete(deptNo);
        console.log('deptCreate result:' + JSON.stringify(response));
        expect(response.status).toEqual('success');
    });

    test('deptByIntegrateId', async () => {
        try {
            // 集成模式才可以调用
            const integrateId = 58335612;
            const response = await api.deptByIntegrateId('58335612');
            console.log('deptByIntegrateId result:' + JSON.stringify(response));
            expect(response['department']['dept_no']).toEqual(integrateId);
        } catch (e) {
            console.log(e);
        }
    });

    test('departmentImport', async () => {
        try {
            const response = await api.departmentImport([{
                parent_no: 1,
                dept_no: 10018,
                name: 'node dept import'
            }]);
            console.log('departmentImport result:' + JSON.stringify(response));
            expect(response.status).toEqual('success');
        } catch (e) {
            console.log(e);
        }
    });
});

File: csharp/test/api/arch/Dept.Test.cs
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/15
*/
using Src.Api.Jdy;
using Src.Base;

namespace Test;

[Collection("Dept")]
[TestCaseOrderer("XUnit.Project.Orderers.AlphabeticalOrderer", "Test")]
public class DeptTest
{
    public static int deptNo = Int32.MinValue;

    private static DeptApiClient api = new DeptApiClient(Global.Host, Global.ApiKey);

    [Fact]
    public async void Test1()
    {
        var dept = await api.deptCreate("Demo研发部门", new Dictionary<string, object>());
        Assert.True(dept != null);
        Console.WriteLine(dept);
        deptNo = dept.Value.GetProperty("department").GetProperty("dept_no").GetInt32();
    }

    [Fact]
    public async void Test2()
    {
        var dept = await api.deptUpdate(deptNo, "Demo测试部门");
        Assert.True(dept != null);
        Assert.Equal(deptNo, dept.Value.GetProperty("department").GetProperty("dept_no").GetInt32());
        Assert.Equal("Demo测试部门", dept.Value.GetProperty("department").GetProperty("name").GetString());
        Console.WriteLine(dept);
    }

    [Fact]
    public async void Test3()
    {
        var option = new Dictionary<string, object>();
        option.Add("hasChild", true);
        var deptList = await api.deptList(deptNo, option);
        Assert.True(deptList != null);
        Console.WriteLine(deptList);
    }

    [Fact]
    public async void Test4()
    {
        var response = await api.deptDelete(deptNo);
        Assert.True(response != null);
        Assert.Equal("success", response.Value.GetProperty("status").GetString());
    }

    [Fact]
    public async void Test5()
    {
        var response = await api.deptByIntegrateId("1005");
        Assert.True(response != null);
        Assert.Equal(3005, response.Value.GetProperty("code").GetInt64());
        Console.WriteLine(response);
    }

    [Fact]
    public async void Test6()
    {
        Dictionary<string, object>[] depts = { };
        var response = await api.departmentImport(depts);
        Assert.True(response != null);
        Assert.Equal(6014, response.Value.GetProperty("code").GetInt64());
        Console.WriteLine(response);
    }
}